workflows/fine_res_budget/budget/budgets.py

```diff
@@ -50,8 +50,10 @@ def pressure_level_average(self, delp_fine, delp_coarse, area, field, factor):
         pi = self.pressure_at_interface(delp_fine)
         pi_c = self.pressure_at_interface(delp_coarse)
         pi_c_up = self.block_upsample(pi_c, factor=factor)
-
-        fg = self.regrid_vertical(pi, field, pi_c_up)
+        if self.z in field.dims:
+            fg = self.regrid_vertical(pi, field, pi_c_up)
+        else:
+            fg = field
```

Comment:
Doing a "pressure level average" of a 2D field doesn't make sense. What about moving this `if` statement one level up and defining a 2D coarsening function that does not take in the pressure thicknesses or call `regrid_vertical`?

Or at the very least, include lines 50-52 inside the `if self.z in field.dims:` block

Reply:
Updated. Moved this up one level, makes a lot more sense.